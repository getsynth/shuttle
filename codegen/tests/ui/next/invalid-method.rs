shuttle_codegen::app! {
    #[shuttle_codegen::endpoint(method = pet, route = "/hello")]
    async fn hello() -> &'static str {
        "Hello, World!"
    }

    #[shuttle_codegen::endpoint(method =, route = "/hello")]
    async fn hello() -> &'static str {
        "Hello, World!"
    }
}
