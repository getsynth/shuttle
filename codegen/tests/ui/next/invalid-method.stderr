error: method is not supported

         = help: Try one of the following: `get`, `post`, `delete`, `put`, `options`, `head`, `trace` or `patch`

 --> tests/ui/next/invalid-method.rs:2:42
  |
2 |     #[shuttle_codegen::endpoint(method = pet, route = "/hello")]
  |                                          ^^^

error: no method provided

         = help: Add a method to your endpoint: `method = get`

 --> tests/ui/next/invalid-method.rs:2:32
  |
2 |     #[shuttle_codegen::endpoint(method = pet, route = "/hello")]
  |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected expression
 --> tests/ui/next/invalid-method.rs:7:41
  |
7 |     #[shuttle_codegen::endpoint(method =, route = "/hello")]
  |                                         ^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/ui/next/invalid-method.rs:11:2
   |
11 | }
   |  ^ consider adding a `main` function to `$DIR/tests/ui/next/invalid-method.rs`
